schemas/com.snowplowanalytics.snowplow/screen_view/jsonschema/1-0-1

```diff
@@ -0,0 +1,61 @@
+{
+	"$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+	"description": "Schema for a screen view event",
+	"self": {
+		"vendor": "com.snowplowanalytics.snowplow",
+		"name": "screen_view",
+		"format": "jsonschema",
+		"version": "1-0-1"
+	},
+	"type": "object",
+	"properties": {
+		"name": {
+			"type": "string",
+            "maxLength": 100,
+            "description": "The name of the screen viewed."
+		},
+		"type": {
+            "type": "string",
+            "maxLength": 100,
+            "description": "The type of screen that was viewed e.g feed / carousel."
+        },
+        "screen_session_id": {
+            "type": "string",
+            "maxLength": 100,
+            "description": "An ID that can be used in lieu of the name property, usually regenerated upon each session."
+        },
+        "screenview_id": {
+            "type": "string",
+            "maxLength": 100,
+            "description": "An ID for identifying a screenview i.e. generated on each screenview."
+        },
+        "previous_screen_name": {
+            "type": "string",
+            "maxLength": 100,
+            "description": "The name of the previous screen."
+        },
+        "previous_screen_session_id": {
+            "type": "string",
+            "maxLength": 100,
+            "description": "A screen session ID of the previous screen."
+        },
+        "previous_screenview_id": {
+            "type": "string",
+            "maxLength": 100,
+            "description": "A screenview ID of the previous screenview."
+        },
+        "previous_type": {
+            "type": "string",
+            "maxLength": 100,
+            "description": "The screen type of the previous screenview."
+        },
+        "transition_type": {
```

Comment:
(Piggybacking off @dilyand 's comment) In my view this is the only property that is really specific to the screen view event, all the other properties can be in the screen entity that we will recommend that all mobile users enable.

Just for my understanding, is this field intended for values such as "button_click" and "swipe"? ie the action that was performed to lead us to this screen?

Reply:
I think everything from `screenview_id` till the end belongs in the event rather than the screen entity. 'Previous' suggests some sort of temporal relationship. That makes sense for events, which follow in chronological order; but not so much for screens, which are not temporally bound.

Reply:
> Just for my understanding, is this field intended for values such as "button_click" and "swipe"? ie the action that was performed to lead us to this screen?

I _think_ transition_type refers to an actual technical aspect of apps. There are different ways of transitioning from one screen to another, and this captures that as far as I know. I'm not sure if they're predicated on the user action like 'button click' or 'swipe', or if they're something like 'View Controller view loaded' vs 'Carousel screen transition' (I made those up but just trying to illustrate)

Reply:
Ah ok makes sense!

Reply:
Don't have much to add here, but figured it'd be good to say I don't have any autotracking for transition type. I'm not completely sure how to accomplish that, and it'd be something that could be added down the line if there's a decent way to go about it.

Reply:
> I think everything from `screenview_id` till the end belongs in the event rather than the screen entity. 'Previous' suggests some sort of temporal relationship. That makes sense for events, which follow in chronological order; but not so much for screens, which are not temporally bound.

Ok - but like I mentioned in another comment, is there any point where the querying is super simplified if you're able to discriminate events by screen details and the previous screen specifics are available right there? I think the intention is making screen transitions really accessible even if it means some redundancy.

Reply:
@mhadam I agree that some tradeoffs would be acceptable if they make working with the data easier. But I think there are no great benefits from redundancy in this case. On BigQuery or Snowflake, we only have a single `atomic.events` table anyway, so we're just duplicating columns in that table. On Redshift, if you use both schemas, you'll still end up with two additional shredded tables, and even if you can maybe do some queries with one join instead of two, I don't think it's worth it. I'm much in favour of keeping things separate and consistent with the concept of events vs entities that we're trying to push in our most recent training materials.

But we should take more opinions on this. @colmsnowplow , @architgoyal1 any definitive thoughts? @chuwy you might want to chime in as well?

Reply:
My two cents: firstly, the context is optional. So IMO everything should remain in the event regardless.

> But I think there are no great benefits from redundancy in this case. On BigQuery or Snowflake, we only have a single atomic.events table anyway, so we're just duplicating columns in that table.  On Redshift, if you use both schemas, you'll still end up with two additional shredded tables, and even if you can maybe do some queries with one join instead of two, I don't think it's worth it.

I see the reasoning here, but I think it's worth highlighting a slightly different perspective - thinking of the benefit less in terms of query efficiency and more in terms of path to value for the user. 

If the data I want to aggregate by is only in one row, I need to account for that before I can gain insights. So I need a 1:many join, a window function, or I need to aggregate then join then aggregate again. If I wasn't proficient at SQL the way you and I are @dilyand, that's not trivial. The simplest option (1:many join) opens the door to problems with duplication, for example. So at the simplest we're asking them to get their head around how to test for duplication in the first step of their query before they get to learning about users.

If that data is in the context though, it's pretty straightforward. I have a 1:1 join then a group by. Hard to get wrong and I'll be confident in the insights I deliver within a few minutes of getting data. 

So IMO the question of what should go in the context should be answered by focusing on what the user is likely going to want to do - I don't think `transition_type` is a likely candidate for aggregations but I do think the rest are.

Reply:
On closer inspection, all of the context fields are already included in the event. What is the purpose of the context then?

Reply:
The event marks a screen view event. The context is attached to all events that happen on screen, like the web page context.

Reply:
In that case why don't we make the same as the web page context? All we need is the `screenview_id`.

Reply:
> If the data I want to aggregate by is only in one row, I need to account for that before I can gain insights. So I need a 1:many join, a window function, or I need to aggregate then join then aggregate again.

If I want to aggregate the events that happen on a screen, then ideally the data I want to aggregate by isn't in only one row.

Reply:
No, I'm saying the context could literally be just the `screenview_id`, attached to every event on the screen, including the `screen_view` event. Just like the web page context works. The data will then be in all rows.

Reply:
Sure. What if I need to aggregate non-screenview events across screen types, or all screen views of the same screen?

The web context works really well for a page views based model, but has these same issues for parallel tasks. That just doesn't come up often since most web analytics starts with page views.

Reply:
Got it.

Reply:
I mean, it's just my opinion here - my initial reaction is that it's valuable/makes life easier. 🙂